src/components/structures/RoomDirectory.js

```diff
@@ -81,20 +81,30 @@ module.exports = React.createClass({
         // });
     },
 
-    showRoom: function(roomId) {
+    showRoom: function(roomIdOrAlias) {
         // extract the metadata from the publicRooms structure to pass
         // as out-of-band data to view_room, because we get information
         // here that we can't get other than by joining the room in some
         // cases.
         var room;
-        for (var i = 0; i < this.state.publicRooms.length; ++i) {
-            if (this.state.publicRooms[i].room_id == roomId) {
-                room = this.state.publicRooms[i];
-                break;
+        if (roomIdOrAlias[0] == '!') {
```

Comment:
I'm probably missing something here, but doesn't this condition prevent the whole function working for aliases?